compiler-rt/lib/asan/asan_rtl.cpp

```diff
@@ -460,6 +492,13 @@ static bool AsanInitInternal() {
   allocator_options.SetFrom(flags(), common_flags());
   InitializeAllocator(allocator_options);
 
+  // Apply ASan flags.
+  // NOTE: In order for options specified through `__asan_default_options` to be
+  // honored on Windows, it is necessary for those options to be configured
+  // inside the `ApplyOptions` method. See the function-level comment for
+  // `ApplyFlags` for more details.
+  ApplyFlags();
```

Comment:
Can this be done after AvoidCVE_2016_2143?

Reply:
Possibly, but only if I also move the allocator initialization block:

https://github.com/llvm/llvm-project/blob/ba740aa73d052e7e68ccdea1d9531bc71eaad53c/compiler-rt/lib/asan/asan_rtl.cpp#L487-L489

Right below `AvoidCVE_2016_2143` as well.

`ApplyFlags` depends on the allocator machinery being initialized, so it has to happen after the fact. I sense it might be safer to keep it in it's current position, but I'm willing to move it if it seems like a non-problem to you.

Reply:
I scoped out the change to `__sanitizer_set_report_path` here -> https://github.com/llvm/llvm-project/pull/122990/commits/a449ffa5a0b9505c21265034a0224f46e1a64943

@vitalybuka please let me know if that seems safer to you. Thanks!